extensions/amp-mraid/0.1/amp-mraid.js

```diff
@@ -109,36 +95,43 @@ export class MraidInitializer {
       this.mraidLoadSuccess_();
     });
     mraidJs.addEventListener('error', () => {
-      this.handleError_(HostServiceError.MISMATCH);
+      if (!this.registeredWithHostServices_) {
+        this.handleMismatch_();
+      }
     });
     const head = document.getElementsByTagName('head').item(0);
     head.appendChild(mraidJs);
   }
 
-  /**
-   * @param {number} hostServiceError
-   */
-  handleError_(hostServiceError) {
-    if (!this.registeredWithHostServices_ &&
-        this.fallbackOn_.includes(hostServiceError)) {
-      this.declineService_();
-    }
-    // TODO: send error ping
-  }
-
   /**
    * Runs when MRAID reports that it is ready.
    */
   mraidReady_() {
```

Comment:
If MRAID just doesn't call this callback then `rejectVisibilityServiceForDoc` won't get called.   How will AMP handle this?  Ideally AMP would:

* never fire any visibility-related events
* ignore expand/close
* fall back to native opener

The last one I think won't happen right now?  Maybe we can fix this with a follow-up that has some sort of timeout?

Reply:
SG, will address in a separate PR.